geopandas/io/tests/test_file.py

```diff
@@ -231,6 +232,34 @@ def test_to_file_datetime(tmpdir, driver, ext, time, engine):
         assert_series_equal(df["b"], df_read["b"])
 
 
+GEOSJON_WITH_MIXED_TIMEZONES = """
+{
+"type": "FeatureCollection",
+"features": [
+{ "type": "Feature", "properties": {
+                    "date": "2014-08-26 10:01:23.040001+02:00" },
+                    "geometry": { "type": "Point", "coordinates": [ 1.0, 1.0 ] } },
+{ "type": "Feature", "properties": {
+                    "date": "2019-03-07 17:31:43.118999+01:00" },
+                    "geometry": { "type": "Point", "coordinates": [ 1.0, 1.0 ] } }
+            ]
+}
+"""
```

Comment:
A little bit simpler, we could actually also create the GeoDataFrame with those values and write that to geojson:

```
df = GeoDataFrame({"date": ["2014-08-26 10:01:23.040001+02:00", "2019-03-07 17:31:43.118999+01:00"]},
    geometry=[Point(1, 1), Point(1, 1)]
)
df.to_json("test_mixed_datetime.geojson")
```

Since geojson has no type information, GDAL will just infer it from the string format anyway whether it sees a datetime field or not, so we can write it as strings.

Reply:
>Alternatively, we could also convert to UTC (pd.to_datetime(.., utc=True)) in case of mixed timezones

Coming back to this, if we do this automatically there is then no way to inspect the original timezones offsets - which is perhaps not an issue, but if you want to re-localise (say into a non-daylight saving time) you have to already know what the offset since you can't inspect it when we read it. 

It also means that with the edge case in #2502 we would have 3 possible output "type" cases (utc datetime, original datetime and object dtype) rather than two.

I'm content just to leave these as notes though, and have added the UTC conversion in the most recent commit.